### PR TITLE
mailto Spam Bot improvement

### PR DESCRIPTION
**Pull request by ziegenberg**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Encode E-Mail Adress and make use of Smarty mailto & encode function for better spam bot protection. See PrestaShop/Prestashop#17562 for reference
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#18928
| How to test?  | Please check in the footer and the contact us page, the contact mail works as before. Now it's protected against SPAM bots

### CODE REVIEW

**Comment by matks** [comment]
Looks good 👍

**Comment by ziegenberg** [comment]
the question is, why the templates here and in themes/classic/modules/ps_contactinfo/ differ substantially?

**Comment by Progi1984** [comment]
@ziegenberg Because contributors forgot to do modifications on modules. But thank you for your PR.

Ready for review ?

**Comment by ziegenberg** [comment]
Yes, ready for review.

If contributors forgot to do modifications on modules, it sounds to me, that the templates here and in themes/classic/modules/ps_contactinfo should not differ. So my next questions: should I file a new PR for updating the templates here to reflect the changes in themes/classic/modules/ps_contactinfo/? I'd gladly do so, if desired.

**Comment by Progi1984** [comment]
If you can, do it :) We review two PR at the same time ;)

**Comment by ziegenberg** [comment]
New PR is in, see #37. It was a bit messy though, as it looks like at some time in the past `ps_contactinfo.tpl` and `ps_contactinfo-rich.tpl` got mixed up.

File: ps_contactinfo-rich.tpl [hunk 1]
```diff
@@ -53,15 +53,7 @@
...
-    {l
-      s='Email us: [1]%email%[/1]'
-      sprintf=[
-        '[1]' => '<span>',
-        '[/1]' => '</span>',
-        '%email%' => $contact_infos.email
-      ]
-      d='Modules.Contactinfo.Shop'
-    }
+    {l s='Email us: ' d='Shop.Theme.Global'}
```
Comment (PierreRambaud) [inline ps_contactinfo-rich.tpl:56]:
```suggestion
    {l s='Email us:' d='Shop.Theme.Global'}
```

File: ps_contactinfo.tpl [hunk 2]
```diff
@@ -50,15 +50,7 @@
...
-      {l
-        s='Email: [1]%email%[/1]'
-        sprintf=[
-          '%email%' => $contact_infos.email,
-          '[1]' => '<a href="mailto:'|cat:$contact_infos.email|cat:'">',
-          '[/1]' => '</a>'
-        ]
-        d='Modules.Contactinfo.Shop'
-      }
+    {l s='Email us: ' d='Shop.Theme.Global'}
```
Comment (PierreRambaud) [inline ps_contactinfo.tpl:53]:
```suggestion
    {l s='Email us:' d='Modules.Contactinfo.Shop'}
```

**Comment by LouiseBonnard** [comment]
Hi there, I don't think there is any rule regarding a wording used in a module that already exists as a back office string (localized in the **Admin**.* domain) but it might the occasion to set one. :-)

@Junebyun, do you confirm? If yes, we will have the two following scenarios:

1. keeping the **Admin**.* domain will allow us to keep the translation and avoid duplicating the string
2. turning the **Admin**.* domain into its own related module domain (**Modules.Contactinfo**.* domain) will make us lose the translation and duplicate the string but allow a more relevant domain (useful for translation and export)

**Comment by ziegenberg** [comment]
I would keep `Modules.Contactinfo.Shop` as in the next PR #37 it's still `Modules.Contactinfo.Shop`. This is probably only copy&paste error on my side from the template files from `themes/classic/modules/ps_contactinfo`

**Comment by Junebyun** [comment]
Hi guys! Correct me if I'm wrong, but would that string appear in the FO? 

Because if so, I found that there is the string: **"Email us: [1]%email%[/1]"** in _Modules.Contactinfo.Shop_ 

Can't we use that one? It would allow us to not duplicate an existing string and it would be in the relevant domain.

**Comment by LouiseBonnard** [comment]
@Junebyun, it is the former wording, you can see it in the red parts (= the removed parts) in the _Files changes_ tab. ;-)

**Comment by Junebyun** [comment]
Right my bad! Then I would go with the second suggestion: _Modules.Contactinfo.Admin_
It might duplicate the string but I prefer that strings are hosted in its own related domain. Plus it seems to be linked to another PR.

**Review by atomiix — APPROVED** [review]

File: ps_contactinfo.tpl [hunk 3]
```diff
@@ -50,15 +50,7 @@
...
-        s='Email: [1]%email%[/1]'
-        sprintf=[
-          '%email%' => $contact_infos.email,
-          '[1]' => '<a href="mailto:'|cat:$contact_infos.email|cat:'">',
-          '[/1]' => '</a>'
-        ]
-        d='Modules.Contactinfo.Shop'
-      }
+    {l s='Email us:' d='Modules.Contactinfo.Shop'}
+    {mailto address={$contact_infos.email} encode="javascript"}
```
Comment (PierreRambaud) [inline ps_contactinfo.tpl:54]:
```suggestion
      {l s='Email us:' d='Modules.Contactinfo.Shop'}
      {mailto address={$contact_infos.email} encode="javascript"}
```

**Review by PierreRambaud — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by Progi1984** [comment]
Hello @ziegenberg , thanks for your contribution, would you mind to rebase your pull request before merging ? Thanks

**Comment by ziegenberg** [comment]
I rebased onto master and pushed.

**Comment by ziegenberg** [comment]
Sorry, that was a mistake. I'll correct this asap.

**Comment by Progi1984** [comment]
@ziegenberg You have files in conflicts. Have you fetched upstream `dev` branch before rebasing ?

**Comment by ziegenberg** [comment]
I fixed it. I originally fetched `master` from upstream, made the stupid mistake and than forgot to fetch `dev` as well. Was a bit in a hurry. Sorry for the inconvenience.

**Pull request closed without merging**